### ISSUE TITLE
Add support for counters

### ISSUE THREAD

**Issue by owentran** [signalfx/golib#51]
We're switching from the https://github.com/zvelo/go-signalfx package to the official signalfx one, but noticed that there is no support for counters, but you do support cumulative counters?

Is there any reason why there is no support for regular counters?

Thanks,
Owen

**Comment by cep21** [comment]
I just never wrote a wrapper function for it.  The wrappers are pretty simple.  It sounds very reasonable to have a wrapper for counters.

**Comment by owentran** [comment]
Thanks for the quick response. I took at datapointcreation.go and it's using `counter` not `cumulative_counter` as the metric type.

https://github.com/signalfx/golib/blob/master/sfxclient/datapointcreation.go#L22
```
// Cumulative creates a SignalFx cumulative counter for integer values.
func Cumulative(metricName string, dimensions map[string]string, val int64) *datapoint.Datapoint {
	return datapoint.New(metricName, dimensions, datapoint.NewIntValue(val), datapoint.Counter, time.Time{})
}
```
Is that a bug?  Or am I completely missing the point on counters vs cumulative counters?

And I have a bonus question. I'm using gauges to record request time for an API call. Works great for tracking the response time. Why can't I just aggregative and do a count on those calls to track the rate of requests instead of using a counter?

**Comment by cep21** [comment]
Resolved in cfd32838437a05719607d47a1d4c4b204c933ab5
